### DISCUSSION TITLE
Duplicate Metrice are sent by openlit

### DISCUSSION THREAD

**Discussion by aravinth-numberz** [openlit/openlit#439] [Q&A]
I'm using openlit for genai tracing

```       
llm_response = ""
with start_trace(name="completions") as trace:
    meta_data = get_trace_meta_data()
    trace._span.set_attributes(meta_data)
    formatted_messages = self.format_messages(messages)
    llm_response = self.model.chat(
        messages=formatted_messages,
        user=meta_data.get("userId"),
    )
    llm_response = str(llm_response).replace("assistant:", "").strip()
return llm_response
```

Once the function is called and got the response, the openlit is sending request log properly to Kibana, But the issue is with usage tokens/cost calculation. After the LLM call is done the calculation is happening and sending it to kibana, but it's not stopping the same log is sent to kibana every minute untill i stop the server(flask).

What could be the reason? can you help me in this
Thanks in advance

**Comment by amanagarwal042** [comment]
Are multiple calls being made to the function? Also are you by chance using Automnatic Instrumentation of OpenLIT aswell (`openlit.init()`)

Also cc @murnitur as this was a change you added and if you have any insights

**Reply by chibuike-okpara** [reply]
It would help if you abstracted that into a function. Also, try calling trace.set_result(RESULT).
Tag me in your response.
